Gemfile

```diff
@@ -61,6 +67,8 @@ group :test do
   gem 'simplecov', require: false
   gem 'cucumber-rails', require: false
   gem 'database_cleaner'
+  gem 'rspec-sidekiq'
```

Comment:
Are we using this anywhere?

In any case, are we able to use the `ActionJob` testing facilities? https://edgeguides.rubyonrails.org/testing.html#testing-jobs

Reply:
I was going to use this but opted for standard have_received matchers at the ```perform_async``` level instead. Will remove

Reply:
About ActionJob, which can work with sidekiq, we'll need a separate decision. Yes, ActiveJob is a more abstract interface and it is part of Rails, but it also limits you to functionality common across all supported queuing systems.

For clockwork-related triggering, I think ActiveJob is overkill, and it'll be easy to switch to ActiveJob if we need to, we'll just grep for ```perform_async``` calls in the code.

Reply:
For testing, if we wanted ActiveJob-like inline execution, Sidekiq supports this (e.g. by adding ```Sidekiq::Testing.inline!``` to spec_helper.rb

I thought about this and it would make sense for some feature tests (when using 'inline' mode, the worker code gets executed immediately at the point of queuing, bypassing any queuing and need for a sidekiq process).

However, for most cases, I think we should be testing a controller action or a scheduled task enqueues a job of the appropriate type (e.g. ClockworkCheck), then test the actual worker itself or its associated service for business logic.